### ISSUE TITLE
🐛ProcessEngine throws error when multiple ETWs poll for tasks at the same time

### ISSUE THREAD

**Issue by Vyperus** [process-engine/process_engine_runtime#396]
## Description

After a medium long discussion with @ElRaptorus I am opening this issue with the purpose to remind my fellow dev of what we found out ;)

It seems like the "ExternalTaskWorkerClient" sometimes sends a "double" authorization header to poll for new tasks for the External Task Worker.

When starting exactly one ETW it seems to work fine. However if we start like >20 ETWs at the same time some of them (sometimes 2 sometimes 3 or even more) seems to break when they start to poll for tasks.

<img width="1666" alt="Bildschirmfoto 2019-09-06 um 10 33 27" src="https://user-images.githubusercontent.com/17248802/64413458-d1c9a800-d091-11e9-8ec9-a7b4c68c7fbc.png">

And the corresponding error thrown by the PE:

<img width="1174" alt="Bildschirmfoto 2019-09-06 um 10 33 41" src="https://user-images.githubusercontent.com/17248802/64413480-d9894c80-d091-11e9-96bf-09e5e7e50fdb.png">

**Comment by ElRaptorus** [comment]
In addition: 
The problem seems to be that the "Authorization" header gets set twice. 
But we do not yet know, if this done by the application sending the http request, or if the http endpoint of the ProcessEngine misinterprets the given `Authorization` header.

Additionally, the `jsonwebtoken` doesn't seem to mind parsing invalid or malformed tokens. It just returns `null`.
The IdentityService then tries to access the `sub` property on that "token", which of course fails with the mentioned error.

I'll try reproducing this, using a large number of ETWs in a sample app, but my gut feeling tells me that the number of ETWs is not the cause of this error.
But even if it is, we still need to robustify the way we handle Auth-headers and tokens.

In general, we can intercept invalid tokens, by checking the result of the `jsonwebtoken` parsing operation. 
Also, we can check the content of the `Authorization` header upon receiving a HTTP request, to see if it contains more than one value.

**Comment by ElRaptorus** [comment]
I tried reproducing this, using a sample app that creates 100 workers with 100 topics; all of which are using the same auth token and run in parallel to each other.
It worked just fine. I did not get any errors.

Pretty sure this issue does not occur at our end. 
I'm going to robustify the IdentityService's error handling and I'll improve the auth-header check of our middlewares, but other than that, I don't think there is much that we can do.
